skydriver/k8s.py

```diff
@@ -0,0 +1,371 @@
+        Failure to do so will trigger an API error.
+
+        Also Containers must be a list!
+
+        Docs3: https://github.com/kubernetes-client/python/issues/589
+        """
+        # Body is the object Body
+        body = kubernetes.client.V1Job(api_version="batch/v1", kind="Job")
+        # Body needs Metadata
+        # Attention: Each JOB must have a different name!
+        body.metadata = kubernetes.client.V1ObjectMeta(namespace=namespace, name=name)
+        # And a Status
+        body.status = kubernetes.client.V1JobStatus()
+        # Now we start with the Template...
+        template = kubernetes.client.V1PodTemplate()
+        template.template = kubernetes.client.V1PodTemplateSpec()
+        # Make Pod Spec
+        template.template.spec = kubernetes.client.V1PodSpec(
+            containers=containers,
+            restart_policy='Never',
+            volumes=[
+                kubernetes.client.V1Volume(
+                    name=n, empty_dir=kubernetes.client.V1EmptyDirVolumeSource()
+                )
+                for n in volumes
+            ],
+        )
+        # And finaly we can create our V1JobSpec!
+        body.spec = kubernetes.client.V1JobSpec(
+            ttl_seconds_after_finished=600, template=template.template
+        )
+        return body
+
+    @staticmethod
+    def create_container(
+        name: str,
+        image: str,
+        env: dict[str, Any],
+        args: list[str],
+        volumes: dict[str, Path],
+    ) -> kubernetes.client.V1Container:
+        """Make a Container instance."""
+        return kubernetes.client.V1Container(
+            name=name,
+            image=image,
+            env=[
+                kubernetes.client.V1EnvVar(name=name, value=value)
+                for name, value in env.items()
+            ],
+            args=args,
+            volume_mounts=[
+                kubernetes.client.V1VolumeMount(name=vol, mount_path=str(mnt))
+                for vol, mnt in volumes.items()
+            ],
+        )
+
+
+class SkymapScannerJob:
+    """Wraps a Skymap Scanner Kubernetes job with tools to start and manage."""
+
+    def __init__(
+        self,
+        api_instance: kubernetes.client.BatchV1Api,
+        rest_address: str,
+        auth_token: str,
+        # docker args
+        docker_tag: str,
+        # condor args
+        njobs: int,
+        memory: str,
+        # scanner args
+        scan_id: str,
+        reco_algo: str,
+        gcd_dir: Path | None,
+        nsides: dict[int, int],
+    ):
+        self.api_instance = api_instance
+
+        # image
+        if not docker_tag:
+            docker_tag = 'latest'
+        image = f"{ENV.SKYSCAN_DOCKER_IMAGE_NO_TAG}:{docker_tag}"
+        self.docker_tag = docker_tag
+
+        # env
+        self.env: dict[str, str | int] = {
+            # broker/mq vars
+            'SKYSCAN_BROKER_ADDRESS': ENV.SKYSCAN_BROKER_ADDRESS,
+            'SKYSCAN_BROKER_AUTH': auth_token,
```

Comment:
This should come from a secret, not as a str in the env.  Same with `SKYSCAN_SKYDRIVER_AUTH`.

Not sure how you'd go about setting `value_from` in `create_container()` to be a [V1EnvVarSource](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1EnvVarSource.md), given your dict types here.  Maybe if you have a nested dict, that's `value_from` instead of `value`?

Reply:
ohhh, I see what you've bee talking about now. I'll figure out how to do that